### DISCUSSION TITLE
Create running Container from Server Dockerfile

### DISCUSSION THREAD

**Discussion by monarc83** [ente-io/ente#736] [Q&A]
Hej all, 

I am currently trying to create a running container from the provided server docker file, but for some unknown reason it already failed at step 5.

Step 4/18 : WORKDIR /etc/ente/
 ---> Using cache
 ---> 523a28eee2e4
Step 5/18 : COPY go.mod .
COPY failed: file not found in build context or excluded by .dockerignore: stat go.mod: file does not exist

Would somebody please be so kind as to help me set up the container successfully. Many thanks in advance.

**Comment by mnvr** [comment]
It sounds like you're not running the commands from the server directory, can you double check you're in the server directory?

This works
```
git clone https://github.com/ente-io/ente
cd ente/server
docker build .
```

---

In general though, you might find it easier to get started by using the provided Docker Compose file. It'll both build the image, and also start a cluster using it.

**Comment by monarc83** [comment]
Oh I missed the docker composed file, so I just run the following and I shoudl be able to connect to the webgui?

git clone https://github.com/ente-io/ente
cd ente/server
docker compose pull
docker compose up -d

**Reply by mnvr** [reply]
Yes, and you don't need the `docker compose pull`. Just `docker compose up --build` is enough. 

After your server is running, you can connect using the web gui from a separate terminal

```
cd ente/web
git submodule update --init --recursive
yarn install
NEXT_PUBLIC_ENTE_ENDPOINT=http://localhost:8080 yarn dev
```

More details are in the [RUNNING.md](https://github.com/ente-io/ente/blob/main/server/RUNNING.md) in the server directory.

**Reply by mnvr** [reply]
EDIT: `docker compose up --build`

**Comment by monarc83** [comment]
until `git submodule update --init --recursive` all went well, I guess, but after installing yarn the final steps aren't working.

```
root@mailtest:/opt/docker/ente/web# yarn install
00h00m00s 0/0: : ERROR: [Errno 2] No such file or directory: 'install'
root@mailtest:/opt/docker/ente/web# NEXT_PUBLIC_ENTE_ENDPOINT=http://localhost:8080 yarn dev
00h00m00s 0/0: : ERROR: [Errno 2] No such file or directory: 'dev'
```

I'm afraid I'm just doing something wrong...

**Reply by mnvr** [reply]
You'll need to use Yarn classic. The above logs seem to be coming from Yarn 4.

**Reply by monarc83** [reply]
how do I install Yarn classic?

currently installed....

```
root@mailtest:/opt/docker/ente/web# yarn --version
0.32
```

**Reply by mnvr** [reply]
Ha, the problem here is opposite, you're on a very old yarn - `0.32` seems like an old version of yarn classic.

Yarn has two versions:
- Yarn classic (currently v1.22.21). This is the stable "yarn", but it is deprecated and meant to be replaced sometime in the future by
- Yarn v4 (but that's currently experimental)

---

Anyways, you need to update your yarn to 1.22.21. I'm not sure how you installed yarn, so I can't tell you the exact steps. The [yarn documentation](https://classic.yarnpkg.com/en/docs/install) might help.

Once you see this,

```
$ yarn --version
1.22.21
```

then please try the `yarn install`.

**Reply by monarc83** [reply]
I've figured it out... I can continue.

**Comment by monarc83** [comment]
next issue...

`error An unexpected error occurred: "https://registry.yarnpkg.com/@mui/icons-material/-/icons-material-5.15.10.tgz: ESOCKETTIMEDOUT".`
